### PR TITLE
Travis fixes

### PR DESCRIPTION
**Pull request by matthewbauer**
## Motivation for this change
- silence Nox building
- tell Nox to build in $HOME/tmp/
## Related issues

issues #12968, #6652

### CODE REVIEW

File: maintainers/scripts/travis-nox-review-pr.sh [hunk 1]
```diff
@@ -44,6 +44,14 @@ while test -n "$1"; do
             nix-shell --packages nixpkgs-lint --run "nixpkgs-lint -f $TRAVIS_BUILD_DIR"
             ;;
 
+        nox)
+            echo "=== Fetching Nox from binary cache"
+
+            # build nox silently so it's not in the log
+            # git is an unlisted dependency of nox
+            nix-build "<nixpkgs>" -A nox -A git > /dev/null
```
Comment (grahamc) [inline maintainers/scripts/travis-nox-review-pr.sh]: Could we fix `nox`'s definition to know about `git`?

**Comment by grahamc** [comment]
I added the WIP label, because it looks like you're still working. Looks good. Make sure to squash the commits at the end though :) :+1:

**Comment by matthewbauer** [comment]
This should be okay to go now.

**Comment by lethalman** [comment]
cc @domenkozar @garbas may know more

**Pull request merged**